api/v1alpha3/azurecluster_types.go

```diff
@@ -48,6 +59,10 @@ type AzureClusterSpec struct {
 	// ones added by default.
 	// +optional
 	AdditionalTags Tags `json:"additionalTags,omitempty"`
+
+	// PrincipalRef is a reference to a principal to be used when reconciling this cluster
+	// +optional
+	PrincipalRef *AzurePrincipalRef `json:"principalRef,omitempty"`
```

Comment:
I think we can simplify this to be the name of the identity to use for this cluster. The identity must reside inside of the cluster namespace and should be owned by the cluster.

All references should be to the known type of AzureIdentity via AAD Pod Identity.